src/components/hoc/withAuth.tsx

```
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react';
+import { useAuth } from '../context/authContext';
+
+const withAuth = (WrappedComponent: React.ComponentType) => {
+    const Wrapper: React.FC = (props) => {
+        const { isAuthenticated, isLoading, token, logout, isLoggingOut } = useAuth();
+
+        const checkAuth = useCallback(() => {
+            console.log('haii');
+            if (!token) {
+                console.log('"gaada token"');
+                isAuthenticated && logout();
```

ESLint

[error] Expected an assignment or function call and instead saw an expression.

(@typescript-eslint/no-unused-expressions)